Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -1191,148 +1181,110 @@ void HTMLMediaElement::update_video_frame_and_timeline()
+        if (track == preferred_video_track && !m_has_selected_preferred_video_track) {
+            enable = TriState::True;
+            m_has_selected_preferred_video_track = true;
+        } else {
+            enable = TriState::False;
         }
+    }
 
-        // 5. If enable is still unknown, then, if the media element does not yet have a selected video track, then set enable to true, otherwise, set
-        //    enable to false.
-        if (enable == TriState::Unknown)
-            enable = m_video_tracks->selected_index() == -1 ? TriState::True : TriState::False;
+    // 5. If enable is still unknown, then, if the media element does not yet have a selected video track, then set enable to true, otherwise, set
+    //    enable to false.
+    if (enable == TriState::Unknown)
+        enable = m_video_tracks->selected_index() == -1 ? TriState::True : TriState::False;
 
-        // 6. If enable is true, then select this track and unselect any previously selected video tracks, otherwise, do not select this video track.
-        //    If other tracks are unselected, then a change event will be fired.
-        if (enable == TriState::True)
-            video_track->set_selected(true);
+    // 6. If enable is true, then select this track and unselect any previously selected video tracks, otherwise, do not select this video track.
+    //    If other tracks are unselected, then a change event will be fired.
+    if (enable == TriState::True)
+        video_track->set_selected(true);
 
-        // NB: See the comment regarding AudioTrack.kind above with regard to https://dev.w3.org/html5/html-sourcing-inband-tracks/.
-        video_track->set_kind(enable == TriState::True ? "main"_utf16 : "translation"_utf16);
+    // NB: See the comment regarding AudioTrack.kind above with regard to https://dev.w3.org/html5/html-sourcing-inband-tracks/.
+    video_track->set_kind(enable == TriState::True ? "main"_utf16 : "translation"_utf16);
 
-        // 7. Fire an event named addtrack at this VideoTrackList object, using TrackEvent, with the track attribute initialized to the new VideoTrack object.
-        TrackEventInit event_init {};
-        event_init.track = GC::make_root(video_track);
+    // 7. Fire an event named addtrack at this VideoTrackList object, using TrackEvent, with the track attribute initialized to the new VideoTrack object.
+    TrackEventInit event_init {};
+    event_init.track = GC::make_root(video_track);
 
-        auto event = TrackEvent::create(realm, HTML::EventNames::addtrack, move(event_init));
-        m_video_tracks->dispatch_event(event);
-    }
+    auto event = TrackEvent::create(realm, HTML::EventNames::addtrack, move(event_init));
+    m_video_tracks->dispatch_event(event);
+}
 
-    if (preferred_video_track.has_value())
-        VERIFY(has_selected_preferred_video_track);
+void HTMLMediaElement::on_metadata_parsed(AK::Duration const& duration)
+{
+    update_volume();
```

Comment:
As a note to me to fix this later, could we add a FIXME to move this to setup_playback_manager? I know it won't work there at the moment, since it only sets the volume if the audio sink exists, but we can make it so that we store volume in PlaybackManager and set it upon creating the mixing sink instead.